platform/arcus-containers/notification-services/src/main/java/com/iris/notification/provider/MailgunEmailProvider.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2019 Arcus Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.iris.notification.provider;
+
+import java.util.Map;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import com.iris.core.dao.AccountDAO;
+import com.iris.core.dao.PersonDAO;
+import com.iris.core.dao.PlaceDAO;
+import com.iris.core.notification.Notifications;
+import com.iris.messages.model.BaseEntity;
+import com.iris.messages.model.Person;
+import com.iris.messages.type.EmailRecipient;
+import com.iris.notification.dispatch.DispatchException;
+import com.iris.notification.dispatch.DispatchUnsupportedByUserException;
+import com.iris.notification.message.NotificationMessageRenderer;
+import com.iris.notification.upstream.UpstreamNotificationResponder;
+import com.iris.platform.notification.Notification;
+import com.iris.platform.notification.NotificationMethod;
+import com.iris.platform.notification.provider.NotificationProviderUtil;
+import com.mailgun.api.v3.MailgunMessagesApi;
+import com.mailgun.client.MailgunClient;
+import com.mailgun.model.message.Message;
+import com.mailgun.model.message.Message.MessageBuilder;
+import com.mailgun.model.message.MessageResponse;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.validator.routines.EmailValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class MailgunEmailProvider implements EmailProvider {
+   private static Logger logger = LoggerFactory.getLogger(MailgunEmailProvider.class);
+   
+   private final PersonDAO personDao;
+   private final PlaceDAO placeDao;
+   private final AccountDAO accountDao;
+   private final MailgunMessagesApi mailgunMessagesApiUS;
+   private final NotificationMessageRenderer messageRenderer;
+   private final UpstreamNotificationResponder responder;
+
+   private final String domain;
+   private final static String SENDER_NAME_SECTION = "sender-name";
+   private final static String SENDER_EMAIL_SECTION = "sender-email";
+   private final static String REPLYTO_EMAIL_SECTION = "replyto-email";
+   private final static String SUBJECT_SECTION = "subject";
+   private final static String PLAINTEXT_BODY_SECTION = "plaintext-body";
+   private final static String HTML_BODY_SECTION = "html-body";
+   private final static EmailValidator EMAIL_VALIDATOR = EmailValidator.getInstance();
+   
```

Comment:
Somewhere in here you need to consume the two injected fields for email "replyto" and "senderemail"

    @Inject @Named("email.sendername") 	private String defaultSenderName;